host/tests/rmfAudio_L3_TestCases/rmfAudio_test01_primaryDataCapture.py

```diff
@@ -0,0 +1,117 @@
+
+    This class uses the `rmfAudioClass` to test primary data capture,
+    downloading necessary test assets, setting up RMF Audio Capture
+    and performing verification of captured audio.
+    """
+    def __init__(self):
+        """
+        Initializes the test class with test name, setup configuration, and sessions for the device.
+
+        Args:
+            None
+        """
+        self.testName  = "test01_primaryDataCapture"
+        super().__init__(self.testName, '1')
+
+    def testAudioCapture(self, file_path:str):
+        """
+        Verifies if the captured audio data is a match with input reference
+
+        Args:
+            file_path (str): File name and path of the output wav file created
+
+        Returns:
+            bool: True if audio capture is a match, False otherwise.
+        """
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+
+        # Download test streams to device
+        url =  test.get("streams")
+        if url is not None:
+            filename = os.path.basename(url[0])
+            ref_file_path = os.path.join(self.cpe.get("target_directory"), filename)
+            return self.testrmfAudio.compareWavFiles(ref_file_path, file_path)
+
+    def testFunction(self):
+        """
+        The main test function that verifies primary data capture with input audio.
+
+        This function:
+        - Starts Primary Data capture.
+        - Captures Audio data on primary interface.
+        - Writes the captured audio data to a wav file.
+        - Compares primary data captured with input audio.
+
+        Returns:
+            bool: Final result of the test.
+        """
+        self.log.testStart(self.testName, '1')
+        result = True
+
+        # 1 for primary data capture (default), 2 for auxiliary data capture.
+        capture_type = 1
+        capture_duration = 10 # data capture duration
+        settings_update = 0 # settings_update=0 (no updates to default settings), 1 (update settings)
+        # For settings update, set settings_update to 1, followed by value for capture_format, Sampling Frequency, FIFO size, threshold. 
+        # Use -1 for settings parameter that you want to retain default value for.
+        # Sample shows 1 for settings update, 0(racFormat_e16BitStereo) for capture format, 5(racFreq_e48000) for sampling rate, 16384 for FIFO and -1 to retain default threshold value.
+        # sample : self.testrmfAudio.updateSettings(capture_type, 1, 0, 5, 16384, -1)
+        test_type = 2 # test_type : 1 for byte counting tests (default), 2 for data tracking tests where audio data is captured.
+        primary_wav_file_name = "/tmp/output_primary.wav"
+
+        self.log.stepStart('Primary audio capture')
+        self.testrmfAudio.openHandle(capture_type)
+        # Retain default Settings in this sample
+        self.testrmfAudio.updateSettings(capture_type, settings_update)

```

Comment:
settings_update already set to 0 in updateSettings do we need to send it again ?
is it making any difference?

Reply:
settings_update is just a variable created to hold value instead of passing 0 directly to updateSettings., 
that variable is being used only once in line 100 : "self.testrmfAudio.updateSettings(capture_type, settings_update)", so updateSettings call is required.